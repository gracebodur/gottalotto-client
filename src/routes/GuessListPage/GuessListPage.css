.title {
    font-size: 2em;
    padding-top: 50px;
}

.pb-link p {
    font-size: 1.5em;
}
