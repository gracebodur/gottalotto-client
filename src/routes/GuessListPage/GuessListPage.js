import React, { Component } from 'react'
import GuessList from '../../components/GuessList/GuessList';
import Nav from '../../components/Nav/Nav';
import GuessesApiService from '../../services/guesses-api-service'
import GuessListContext from '../../contexts/guessListContext'
import './GuessListPage.css'

import Scroll from '../../components/Scroll/Scroll'
import Typical from 'react-typical'

class GuessListPage extends Component {
	static contextType = GuessListContext;

	componentDidMount() {
		this.context.clearError();
		GuessesApiService.getAllGuesses()
			.then(this.context.setGuessList)
			.catch(this.context.setError);
	}

	render() {
		return (
			<div>
				<header>
					<Nav />
				</header>
				<Scroll>
				<Typical
					steps={["Everyone else's guesses!", 1000, "", 500]}
					loop={Infinity}
					wrapper="h1"
					className='title'
			  	/>
					<GuessList />
				<a href='https://www.powerball.com/games/home' className='pb-link'><p>Power Ball Official Website</p></a>
				</Scroll>
			</div>
		)
	}
}

export default GuessListPage