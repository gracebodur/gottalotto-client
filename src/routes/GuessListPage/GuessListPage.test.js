import React from "react";
import toJson from "enzyme-to-json";
import GuessListPage from "./GuessListPage";

describe(`GuessListPage component`, () => {
	const props = {
		id: "a",
		name: "test-class-name"
	};

	it("renders a GuessListPage by default", () => {
		const wrapper = <GuessListPage />;
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("renders the GuessListPage given props", () => {
		const wrapper = <GuessListPage {...props} />;
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});