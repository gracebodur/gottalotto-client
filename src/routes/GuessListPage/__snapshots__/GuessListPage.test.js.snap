// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuessListPage component renders a GuessListPage by default 1`] = `undefined`;

exports[`GuessListPage component renders the GuessListPage given props 1`] = `undefined`;
